src/components/Compounder/CompounderVaultDepositForm.tsx

```diff
@@ -0,0 +1,19 @@
+import { FC } from "react"
+
+import { CompounderVaultProps, ProductType } from "@/lib/types"
+import { useVault } from "@/hooks"
+
+import { VaultDepositForm } from "@/components/VaultRow/lib"
+
+export const CompounderVaultDepositForm: FC<CompounderVaultProps> = (props) => {
+  const vault = useVault(props)
+  const underlyingAssets = vault.data?.underlyingAssets
+  const args = {
+    ...props,
+    inputToken: props.asset,
+    outputToken: props.vaultAddress,
+    underlyingAssets,
+    productType: "compounder" as ProductType,
+  }
+  return <VaultDepositForm {...args} />
```

Comment:
I'd probably just directly pass the props without creating variables:

```tsx
return (
  <VaultDepositForm
    {...props}
    inputToken={props.asset}
    outputToken={props.vaultAddress}
    underlyingAssets={vault.data?.underlyingAssets}
    productType="compounder"
  />
)
```

Reply:
What's the rules here? When we should/have to create a new interface and when we shouldn't/haven't?